saleor/product/models.py

```diff
@@ -506,6 +466,42 @@ def __str__(self):
         return self.name or str(self.product_variant)
 
 
+class ProductVariantChannelListingQueryset(models.QuerySet):
+    def create(self, **kwargs):
+        """Create a product's variant channel listing.
+
+        After the creation update the "discounted_price" of the product.
+        """
+        variant_channel_listing = super().create(**kwargs)
+
+        from .tasks import update_product_discounted_price_task
+
+        update_product_discounted_price_task.delay(
+            variant_channel_listing.variant.product_id
+        )
+        return variant_channel_listing
+
+    def bulk_create(self, objs, batch_size=None, ignore_conflicts=False):
+        """Insert each of the product's variant channel listings instances into the database.
+
+        After the creation update the "discounted_price" of all the products.
+        """
+        variants = super().bulk_create(
+            objs, batch_size=batch_size, ignore_conflicts=ignore_conflicts
+        )
+        product_ids = set()
+        for obj in objs:
+            product_ids.add(obj.variant.product_id)
+        product_ids = list(product_ids)
+
+        from .tasks import update_products_discounted_prices_of_catalogues_task
+
+        update_products_discounted_prices_of_catalogues_task.delay(
+            product_ids=product_ids
+        )
+        return variants
+
+
```

Comment:
I'm thinking about dropping this code. IMO we should explicitly call tasks to update discounted prices in places where we know we need to do this (which we already do in mutations for example). I'm afraid that calling tasks automatically in ORM methods might bite us and e.g. trigger these tasks too often.

Reply:
I wholeheartedly agree with @maarcingebala.

Reply:
Let's remove this and double-check that we update discount prices when creating or updating variants (variant channel listings).

Reply:
This may be a good opportunity to generate some charts with mermaid 😉